### ISSUE TITLE
Login shows "connect account" even after accounts are already connected.

### ISSUE THREAD

**Issue by pranavgore09** [fabric8io/fabric8-ui#956]
Steps to reproduce
1. Visit https://prod-preview.openshift.io
2. Log in using RHD account
3. You see following page
![connect_accounts](https://cloud.githubusercontent.com/assets/2026565/25264237/2bc282a2-2683-11e7-8390-f50f857aa727.png)
This page reappear even after connecting accounts by submitting above form.

And my accounts are already linked on `/auth/realms/fabric8/account/identity`
![linked_accounts](https://cloud.githubusercontent.com/assets/2026565/25264246/38dcc5ec-2683-11e7-8db9-acf68d3e7c9e.png)
Maybe I am the only one facing this. not sure if this is a bug or some wrong configuration for my account.

**Comment by joshuawilson** [comment]
I can think of several things here. 
 - What is the link you used to connect? It might be unrelated to what is used by prod-preview, maybe. @alexeykazakov please confirm.
 - The getting started page is designed to connect you when you first login and it does not expect you to connect yourself through a backdoor. We you able to get past the screens? If so, I don't see this as a high priority. I don't expect too many will use the account linking backdoor.

**Comment by alexeykazakov** [comment]
UI should check if account already linked. It's just one call per link. I assumed UI already does it. No matter if it's first login or not. This is something wrong with that page. Username is empty.
And I can confirm that Pranavgore09 is already linked in sso.prod-preview

**Comment by sbose78** [comment]
```
Router Event: NavigationEnd
patternfly.css:2 NavigationEnd(id: 1, url: '/_gettingstarted', urlAfterRedirects: '/_gettingstarted')
patternfly.css:2 NavigationEndid: 1url: "/_gettingstarted"urlAfterRedirects: "/_gettingstarted"__proto__: Object
patternfly.css:2 ERROR TypeError: Cannot read property 'split' of undefined
    at ProviderService.parseJwt (7.8e079b0….chunk.js:1)
    at ProviderService.link (7.8e079b0….chunk.js:1)
    at ProviderService.linkAll (7.8e079b0….chunk.js:1)
    at GettingStartedComponent.connectAccounts (7.8e079b0….chunk.js:1)
    at Object.eval [as handleEvent] (GettingStartedComponent.ngfactory.js:1224)
    at Object._ [as handleEvent] (patternfly.css:2)
    at Object.handleEvent (patternfly.css:2)
    at dispatchEvent (patternfly.css:2)
    at patternfly.css:2
    at HTMLButtonElement.<anonymous> (patternfly.css:2)
    at ZoneDelegate.invokeTask (polyfills.f49acb8….bundle.js:1)
    at Object.onInvokeTask (patternfly.css:2)
    at ZoneDelegate.invokeTask (polyfills.f49acb8….bundle.js:1)
    at Zone.runTask (polyfills.f49acb8….bundle.js:1)
    at HTMLButtonElement.ZoneTask.invoke (polyfills.f49acb8….bundle.js:1)
defaultErrorLogger @ patternfly.css:2
```

this is the trace in the browser console.

**Comment by sbose78** [comment]
I'm locked out now - no amount of refresh/incongnito is helping me get it. :(

**Comment by pranavgore09** [comment]
@joshuawilson , I do not intend to link the account, but UI does not let me in without connecting account hence blocked. And in my case "connect account" does not do anything but takes me to same page. Hence it has become a blocker at least for me on prod-preview.
@sbose78 : I hanvn't seen that error yet :smile: console is clean.

**Comment by dlabrecq** [comment]
I could not reproduce your issue. I was able to log into https://prod-preview.openshift.io just fine. I had to register a new account, so I didn't have GitHub and OpenShift linked. However, I also removed the links from keycloak to try again. In both cases, the getting started page linked my accounts and redirected me to the home page.

You might also try removing your GitHub and OpenShift link from keycloak? Although, if you don't see a username, something else may be going on. For example, you might have a stale cache?

Regardless, you can also type https://prod-preview.openshift.io/_home and bypass the getting started page.

**Comment by dlabrecq** [comment]
FYI, I just checked with Alask and the getting started page is also working fine for him.

**Comment by dlabrecq** [comment]
I was able to reproduce this only when the network went down and I navigated directly to the getting started page. In this scenario, I wasn't logged in properly. Thus, the username field was blank and GitHub and OpenShift were unchecked.

That said, I've added AuthGuard to ensure the user is logged in. If there is a problem with the login, the user should never get here, at least.

See:
https://github.com/fabric8io/fabric8-ui/pull/986
